webrtc/RTCIceCandidate-constructor.html

```diff
@@ -0,0 +1,178 @@
+<!doctype html>
+<title>RTCIceCandidate constructor</title>
+<script src=/resources/testharness.js></script>
+<script src=/resources/testharnessreport.js></script>
+<script>
+  'use strict';
+
+  const candidateString = 'candidate:1905690388 1 udp 2113937151 192.168.0.1 58041 typ host generation 0 ufrag thC8 network-cost 50'
+  const randomString = '<arbitrary string[0] content>;'
+
+  test(t => {
+    // The argument for RTCIceCandidateInit is optional (w3c/webrtc-pc#1153),
+    // but the constructor throws because both sdpMid and sdpMLineIndex are null
+    // Note that current browsers pass this test but may throw TypeError for
+    // different reason, i.e. they don't accept empty argument.
+    // Further tests below are used to differentiate the errors.
+    assert_throws(new TypeError,
+      () => new RTCIceCandidate());
+  }, 'new RTCIceCandidate()');
+
+  test(t => {
+    // All fields in RTCIceCandidateInit are optional,
+    // but the constructor throws because both sdpMid and sdpMLineIndex are null
+    // Note that current browsers pass this test but may throw TypeError for
+    // different reason, i.e. they don't allow undefined candidate string.
+    // Further tests below are used to differentiate the errors.
+    assert_throws(new TypeError,
+      () => new RTCIceCandidate({}));
+  }, 'new RTCIceCandidate({})');
+
+  test(t => {
+    // Check that manually filling the default values for RTCIceCandidateInit
+    // still throws because both sdpMid and sdpMLineIndex are null
+    assert_throws(new TypeError,
+      () => new RTCIceCandidate({
+        candidate: '',
+        sdpMid: null,
+        sdpMLineIndex: null,
+        ufrag: undefined
+      }));
+  }, 'new RTCIceCandidate({ ... }) with manually filled default values');
+
+  test(t => {
+    // Throws because both sdpMid and sdpMLineIndex are null
+    assert_throws(new TypeError,
+      () => new RTCIceCandidate({
+        candidate: ''
+      }));
+  }, `new RTCIceCandidate({ candidate: '' })`);
```

Comment:
> My interpretation based on WebIDL spec is that they are indeed default to null if unspecified. All fields in a dictionary is optional unless specified. The fields with default values would convert undefined to the default value. (https://heycam.github.io/webidl/#dfn-dictionary)

Yes, that's not in contention. My concern is really: why are they nullable, not `required`, but will throw if both are not present and defaulting to null, or both are explicitly null. I think the spec itself needs to address this. 

> Using this we can then point out flaws in the specs using these tests and then improve/clarify the spec if necessary.

Ok, I agree with this.